auth/template.yaml

```diff
@@ -970,6 +970,35 @@ Resources:
         - Key: System
           Value: Authentication
 
+  AuthProxyWafAlarm:  
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub ${AWS::StackName}-auth-proxy-waf-rate-limit
+      AlarmDescription: "Alarm when the Auth Proxy WAF rate limit exceeds 300 requests per 5 minutes"
+      Namespace: AWS/WAFV2
+      ActionsEnabled: true
+      MetricName: AuthWAFErrorRate
+      Statistic: Sum
+      Period: 300
+      EvaluationPeriods: 5
+      Threshold: 300
+      DatapointsToAlarm: 5
```

Comment:
It would be worth adding some test evidence to demonstrate that the alert fires if these conditions are met